File: extract-video-data/transcriptions.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY});

  const response = await client.index.video.transcription(
    '66e9358a808d95368f6f7a7c',
    '66ebe4bfd4f452b062c63e3a',
)

  response.forEach((transcript, index) => {
    console.log(`\nTranscript ${index + 1}:`);
    console.log(`  Start time: ${transcript.start} seconds`);
    console.log(`  End time: ${transcript.end} seconds`);
    console.log(`  Transcription: ${transcript.value}`);
  });
})();

File: extract-video-data/transcriptions.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

# index: 66e9358a808d95368f6f7a7c
# video: 66ebe4bfd4f452b062c63e3a

response = client.index.video.transcription(
    index_id="66e9358a808d95368f6f7a7c",
    id="66ebe4bfd4f452b062c63e3a",
    start=100,
    end=300
)

for i, transcript in enumerate(response, start=1):
    print(f"\nTranscript {i}:")
    print(f"  Start time: {transcript.start} seconds")
    print(f"  End time: {transcript.end} seconds")
    print(f"  Transcription: {transcript.value}")
